src/components/contentLibrary/contentLibrary.component.tsx

```diff
@@ -0,0 +1,95 @@
+import clsx from "clsx";
+import { m, useMotionValue } from "framer-motion";
+import React, { useEffect, useState, useRef } from "react";
+
+import { ObjectList } from "src/components/objectListing";
+import { Panel } from "src/components/panel";
+
+export const ContentLibrary = () => {
+  const [activePanelObject, setActivePanelObject] = useState<{
+    objectType: string;
+    uid: string;
+  } | null>(null);
+
+  const [windowSize, setWindowSize] = useState(0);
+  const objectListingWidth = useMotionValue<number | undefined>(undefined); // from localstorage get saved sized
+  const objectListingRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    const updateWindowSize = () => {
+      setWindowSize(window.innerWidth);
+      if (!activePanelObject) objectListingWidth.set(undefined);
+    };
+    updateWindowSize();
+
+    window.addEventListener("resize", updateWindowSize);
+    return () => {
+      window.removeEventListener("resize", updateWindowSize);
+    };
+  }, [activePanelObject, objectListingWidth]);
+
+  const handleDrag = React.useCallback(
+    (event: unknown, info: { delta: { x: number } }) => {
+      const width =
+        objectListingWidth.get() || objectListingRef?.current?.offsetWidth || 0;
+      const newWidth = width + info.delta.x;
+      if (newWidth > 425 && newWidth < windowSize - 375) {
+        objectListingWidth.set(newWidth);
+      }
+    },
+    [objectListingWidth, windowSize],
+  );
+
+  const objecListingSize = clsx(
```

Comment:
Move this directly into the `classname` on line 57 so all the class is together

Reply:
good spot